src/devtools/views/Components/TreeContext.js

```diff
@@ -127,6 +129,24 @@ function reduceTreeState(store: Store, state: State, action: Action): State {
           selectedElementIndex = null;
         }
         break;
+      case 'SELECT_CHILD_ELEMENT_IN_TREE':
+        if (selectedElementIndex !== null) {
+          const selectedElement = store.getElementAtIndex(
+            ((selectedElementIndex: any): number)
+          );
+          if (
+            selectedElement !== null &&
+            selectedElement.children.length > 0 &&
+            !selectedElement.isCollapsed
```

Comment:
I don't think we need this check, and maybe we should remove it to be more consistent with the other actions here. (The layout effect below will ensure that if we were to select a child within a collapsed subtree, the full tree above the child would be expanded.)

```diff
diff --git a/src/devtools/views/Components/TreeContext.js b/src/devtools/views/Components/TreeContext.js
index d610cdd..d40802f 100644
--- a/src/devtools/views/Components/TreeContext.js
+++ b/src/devtools/views/Components/TreeContext.js
@@ -136,12 +136,14 @@ function reduceTreeState(store: Store, state: State, action: Action): State {
           );
           if (
             selectedElement !== null &&
-            selectedElement.children.length > 0 &&
-            !selectedElement.isCollapsed
+            selectedElement.children.length > 0
           ) {
             const firstChildID = selectedElement.children[0];
             const firstChildIndex = store.getIndexOfElementID(firstChildID);
             if (firstChildIndex !== null) {
+              lookupIDForIndex = false;
+
+              selectedElementID = firstChildID;
               selectedElementIndex = firstChildIndex;
             }
           }
```

Thoughts?

Reply:
I only added this check to fix the bug I mentioned to you earlier. But as you see from next commit that bug might have been caused by a missing dependency. I’ll double check.

Reply:
Ah, ok. I tested the diff above and it seemed okay 😄

Reply:
The bug only showed up on large and complex internal trees.

Reply:
Intriguing.